backend/src/v5/processors/teamspaces/projects/models/commons/tickets.groups.js

```diff
@@ -40,14 +57,23 @@ const getObjectArrayFromRules = async (teamspace, project, model, revId, rules)
 		}
 	}
 
-	const [
-		{ matched, unwanted },
-		idToMeshes,
-	] = await Promise.all([
-		getMetadataByRules(teamspace, project, model, revision, rules, { parents: 1 }),
-		getIdToMeshesMapping(teamspace, model, revision),
-	]);
+	const projection = { parents: 1,
+		...(convertTo3dRepoIds ? {} : { metadata: {
+			$elemMatch: { $or: Object.values(externalIdNamesToKeys).flat().map((n) => ({ key: n })) } } }) };
 
+	const { matched, unwanted } = await getMetadataByRules(teamspace, project, model, revision, rules, projection);
+
+	if (!convertTo3dRepoIds && matched.some((m) => m.metadata)) {
```

Comment:
Bit confused with this logic.

so if we don't want 3drepo Ids, and there are no match, it will fall out of this function and the IdToMeshes stuff?

Reply:
its basically saying that if convertIds is false and there are any matched metadata return the external Ids otherwise return 3d repo Ids

Reply:
@ChristosTsiotsias  this is my issue.

If the user has asked for ifc guids etc, the schema we're returning shouldn't contain 3drepo ids just because we couldn't find any matches. if should return an empty array for ifc_guids or something.